tests/desugar_datatypes/rowPromote.links

```diff
@@ -0,0 +1,5 @@
+typename PolyPingPong =
+  [+|
+    Go: forall a::Row. !a.?a.PolyPingPong,
```

Comment:
Or, this example should be fixed to be `!(a).?(a).PolyPingPong` that is to actually convert the row variables into types. (as in #491).  But we should not be promoting row variables to type variables no matter what.

Reply:
Hrmr - from what I can tell, there shouldn't be a difference between `!(a).X` and `!a.X` any more, so this should fail in both cases (you'd need an explicit `(a|)`). Maybe getting the wrong end of the stick though!

I've just deleted this test case for the time being. It was ignored, and I don't think it really checks anything which wasn't done elsewhere in the suite.

Reply:
I think you're right, I hadn't reread the later part of the issue where I (re)learned the harsh lesson that `(a)` does not turn a row into a type like I thought.  (I think you mean `(|a)` though.)

As long as no Row->Type implicit coercion was happening (or intended) I'm happy now!